bin/ember-template-lint.js

```diff
@@ -49,15 +49,24 @@ function lintSource(linter, options, shouldFix) {
   }
 }
 
-function expandFileGlobs(filePatterns, ignorePattern) {
+function executeGlobby(pattern, ignore) {
+  return (
+    globby
+      // `--no-ignore-pattern` results in `ignorePattern === [false]`
+      .sync(pattern, ignore[0] === false ? {} : { ignore, gitignore: true })
+      .filter((filePath) => filePath.slice(-4) === '.hbs')
+  );
+}
+
+function expandFileGlobs(filePatterns, ignorePattern, glob = executeGlobby) {
   let result = new Set();
 
   filePatterns.forEach((pattern) => {
     let isHBS = pattern.slice(-4) === '.hbs';
-    let isLiteralPath = !isValidGlob(pattern) && fs.existsSync(pattern);
+    let isLiteralPath = !isGlob(pattern) && fs.existsSync(pattern);
 
     if (isHBS && isLiteralPath) {
-      let isIgnored = !micromatch.isMatch(pattern, ignorePattern);
```

Comment:
This logic was just wrong, this is explicitly matching against the ignore pattern so it should not be negated.